samples/drivers/pwm/README.rst

```diff
@@ -0,0 +1,31 @@
+.. _pwm_test:
```

Comment:
There are already PWM samples and tests in the tree. Please have a look at the following instead of introducing a new sample:
- `samples/basic/blinky_pwm`
- `samples/basic/fade_led`
- `samples/basic/rgb_led`
- `tests/drivers/pwm/`